fix(swarm): keep connections alive while active streams exist (#4595)
paritytech/rust-libp2p: swarm/CHANGELOG.md

> ## 0.43.6
> - Keep connections alive whilst there are active streams and implement `no_keep_alive` for `Stream`

Reviewer: ```suggestion
## 0.43.6 - unreleased

- Default to keeping connections alive whilst there are active streams and add `Stream::no_keep_alive` to opt-out of this behaviour.
```

Author: fix document in changelog  412aa75

  See [PR 4225](https://github.com/libp2p/rust-libp2p/pull/4225).
- Remove deprecated `keep_alive_timeout` in `OneShotHandlerConfig`.
  See [PR 4677](https://github.com/libp2p/rust-libp2p/pull/4677).
- Simplify the connection keep-alive algorithm.
  Connections are now kept alive by default if there are active `Stream`s on it.

>   As a result, the `ConnectionHandler::connection_keep_alive` method now has a default implementation that returns `false`.

Reviewer 2: ```suggestion
  In case of no active `Stream`s the `ConnectionHandler` is asked whether it wants to keep the connection alive via  `ConnectionHandler::connection_keep_alive` which defaults to `false`.
```

Would be easier for me to understand, more specifically that `connection_keep_alive` is only called in case there are no active streams.

Author: updated 92823e1

>   See its documentation for more details.

Reviewer 2: ```suggestion
  See documentation on `ConnectionHandler::connection_keep_alive` for more details.
```

Author: updated 92823e1

  See [PR 4225](https://github.com/libp2p/rust-libp2p/pull/4225).
- Remove deprecated `keep_alive_timeout` in `OneShotHandlerConfig`.
  See [PR 4677](https://github.com/libp2p/rust-libp2p/pull/4677).
- Add `ConnectionHandler::connection_keep_alive` default implementation that returns `KeepAlive::No`.

>   See [PR 4703](https://github.com/libp2p/rust-libp2p/pull/4703).

Reviewer: Needs to mention the rename as well :)

Author: updated 3d10eb4
